Folder Computation documentation needs update #7657

ShrushtiDesai opened the pull request:
This PR addresses the issue #7657.

This pull request updates the **Folder Computation** section of the Jenkins documentation to reflect recent UI changes in the handling of computed folders for both **Multibranch Pipeline** Projects and **GitHub Organization** Projects.

#### Key Updates:
- The revised section now describes the previous functionality of **Folder Computation** and **Branch Indexing**, and clarifies the renaming and relocation of these options in the updated Jenkins version:
  - **Multibranch Pipeline Projects**: The **Branch Indexing** tab, which included **Run Now** and **Log** options, has been renamed to **Scan Multibranch Pipeline Now** and **Scan Multibranch Pipeline Log**.
  - **GitHub Organization Projects**: The **Folder Computation** tab, which included **Run Now** and **Log** options, has been renamed to **Scan Organization Now** and **Scan Organization Log**. 
  - These changes followed the deprecation of the **GitHub Organization Folder** plugin and its integration into the **Branch Source** plugin.

- Screenshots and their corresponding alt text have been updated to reflect the UI changes described in this section.

Please review the changes and let me know if further revisions are needed.

kmartens27 reviewed (approved):
This change looks great and I appreciate that the Folder Computation header was kept (as this avoids any link breakage). Thank you very much @ShrushtiDesai for your work on this and for collaborating to ensure the updates are done properly.

Approving pending the format adjustments (nothing major, just a best practice)

timja commented on content/doc/book/pipeline/pipeline-as-code.adoc:
```diff
@@ -80,34 +80,23 @@ dropdowns in other Jenkins jobs.
-
+_Multibranch Pipeline_ projects and _GitHub Organization Folders_ extend Jenkins' folder functionality by introducing computed folders, which automatically manage their contents by dynamically creating and updating child items.
+ 
+* **Multibranch Pipeline Projects:** Previously, the computed folder functionality was located under the **Branch Indexing** tab, which included **Run Now** and **Log** options. Jenkins would generate child items for each eligible branch within a repository and automatically update the pipeline list as branches changed. In the updated version, these options are no longer under a specific tab and  ...
```
please do not include documentation for content that is many years out of date, someone reading the docs doesn't need to know information this old (looks like 8 years ago)

timja commented on content/doc/book/pipeline/pipeline-as-code.adoc:
```diff
@@ -80,34 +80,23 @@ dropdowns in other Jenkins jobs.
+
+* **GitHub Organization Projects:** Originally, the computed folder functionality appeared under the **Folder Computation** tab, with **Run Now** and **Log** options. This functionality populated repositories as individual Multibranch Pipelines and provided insights into branch and repository scans. In the updated version, these options are no longer under a tab and have been renamed to **Scan Or ...
+ 
+These updates followed the deprecation of the plugin:github-organization-folder[*GitHub Organization Folder*] plugin and its integration into the plugin:github-branch-source[*GitHub Branch Source*] plugin.
```
delete

ShrushtiDesai replied:
Hi @timja,
Do you prefer the content to exclude information about the previous UI and the plugin deprecation, and instead focus solely on describing the current UI? Let me know your thoughts so I can adjust the content accordingly.

timja replied:
Yes please current UI only, unless in a transition period not 8 years ago

kmartens27 commented:
Thank you very much @timja for the insights and feedback, I'll be keeping all that in mind going forward as well.

ShrushtiDesai commented:
According to the Feedback from @timja and @kmartens27 i have updated the Folder Computation as follows:
- Removed mentioned of previous UI features.
- Made the UI terms bold.
- Adjusted the formatting as one sentence per line.
- Wording adjustment to avoid past tense.

I have a couple of clarifications:
- Regarding the formatting change to have one sentence per line: I noticed that this formatting is reflected in the code file but not on the UI page itself. Is this what you intended—having the change only in the code for better readability, without affecting the UI?
- Should the deprecation sentence also be removed, or is it okay to keep it as is?
- After making the changes, I see an option to force push the updates. Is it fine to proceed with a force push in this case?

kmartens27 commented:
Hi @ShrushtiDesai, thank you very much for taking the time to make the necessary changes.

In terms of your questions:

1. The practice of one sentence per line makes it easier to review and provide feedback on anything that is written. For instance, if an entire paragraph is on 1 line and there is an issue with something in the middle of the content, it is more cumbersome to provide this feedback. The UI is able to render the line breaks and formatting properly, without having to provide a massive text block. This is indeed by design.
2. Line 89 can be removed, as @timja pointed out this transition happened long ago so it's fine to just focus on the current UI and functionality.
3. Do not force push the changes. Make sure you fetch & pull all the changes down to your working branch and once you've done that you should be able to just push (without the force).
